ss_rules.json

```diff
@@ -0,0 +1,792 @@
+					"FM Title (fmt)"
+				]
+			}
+		}
+	},
+	"Section-Foreword (sfw)": {
+		"section_type":	"frontmatter",
+		"contiguous_block_criteria_01": {
+			"multiple":	"False",
+			"styles": [
+				"FM Head (fmh)",
+				"FM Head ALT (afmh)",
+				"FM Head Nonprinting (fmhnp)",
+				"FM Title (fmt)"
+			],
+			"first_child": {
+				"text": [
+					"Foreword"
+				],
+				"match":	"True"
+			},
+			"previous_sibling": {
+				"required_styles": [
+					"Section-Ad-Card (sac)",
+					"Section-Front-Sales (sfs)",
+					"Section-Series-Page (ssp)",
+					"Section-Halftitle (sht)",
+					"Section-Titlepage (sti)",
+					"Section-Copyright-Page (scr)",
+					"Section-Dedication (sde)",
+					"Section-Contents (stc)",
+					"Section-Introduction-Frontmatter (sif)",
+					"Section-Preface (spf)",
+					"Section-Foreword (sfw)",
+					"Section-Introduction-Chapter (sic)",
+					"Section-Part (spt)",
+					"Section-Chapter (scp)",
+					"Section-Appendix (sap)",
+					"Section-Acknowledgments (sak)",
+					"Section-Afterword (saw)",
+					"Section-Conclusion (scl)",
+					"Section-About-Author (saa)",
+					"Section-Back-Ad (sba)",
+					"Section-Epigraph (sep)",
+					"Section-Excerpt-Opener (seo)",
+					"Section-Excerpt-Chapter (sec)",
+					"Section-Notes (snt)",
+					"Section-Bibliography (sbi)",
+					"Section-Glossary (sgl)",
+					"Section-Index (sin)"
+				]
+			}
+		}
+	},
+	"Section-Introduction-Chapter (sic)": {
+		"section_type":	"main",
+		"contiguous_block_criteria_01": {
+			"multiple":	"True",
+			"styles": [
+				"Chap Title (ct)",
+				"Chap Title ALT (act)",
+				"Chap Title Nonprinting (ctnp)"
+			],
+			"first_child": {
+				"text": [
+					"Introduction"
+				],
+				"match":	"True"
+			},
+			"previous_sibling": {
+				"required_styles": [
+					"Section-Introduction-Chapter (sic)"
+				]
+			}
+		}
+	},
+	"Section-Part (spt)": {
+		"section_type":	"main",
+		"contiguous_block_criteria_01": {
+			"multiple":	"True",
+			"styles": [
+				"Part Contents Head (pch)",
+				"Part Contents Main Head (pcmh)",
+				"Part Epigraph - non-verse (pepi)",
+				"Part Epigraph - verse (pepiv)",
+				"Part Epigraph Source (peps)",
+				"Part Number (pn)",
+				"Part Opening Text (potx)",
+				"Part Opening Text No-Indent (potx1)",
+				"Part Ornament (porn)",
+				"Part Ornament ALT (porn2)",
+				"Part Subtitle (pst)",
+				"Part Title (pt)"
+			],
+			"previous_sibling": {
+				"required_styles": [
+					"Section-Part (spt)"
+				]
+			}
+		}
+	},
+	"Section-Chapter (scp)": {
+		"section_type":	"main",
+		"contiguous_block_criteria_01": {
+			"multiple":	"True",
+			"styles": [
+				"Chap Number (cn)",
+				"Chap Title (ct)",
+				"Chap Title Nonprinting (ctnp)"
+			],
+			"previous_until": {
+				"required_styles": [
+					"Section-Chapter (scp)"
+				],
+				"previous_until_stop": [
+					"Chap Number (cn)",
+					"Chap Title (ct)",
+					"Chap Title Nonprinting (ctnp)",
+					"Titlepage Book Title (tit)"
+				]
+			}
+		},
+		"section_required": {
```

Comment:
I'm not totally clear on what's happening here either. My initial expectation was to see a "section_required: true" value or something along those lines. What is insert_before doing?